### ISSUE TITLE
UPenn's race & ethnicity data

### ISSUE THREAD

**Issue by Joe-Hilgard** [raklein/ml4#4]
UPenn inhouse provides race data as a series of numeric codes. I'm not sure whether these numeric codes match the in-house Qualtrics form or whether they've been recoded to match the race codes used across all sites.

If the codes match the provided in-house Qualtrics form, then I am a little surprised by Wharton's strong Hispanic representation and small Asian representation: 29 Hispanic, 14 Black, 30 White, 6 Native American, 7 Asian/Pacific Islander.

If the codes match the race codes used across all sites, then I am surprised by the strong representation of Native Americans and Native Hawaiians and the small Asian representation: 29 White, 14 Black, 30 Native American (13 of which born outside the USA), 6 Asian, 5 Native Hawaiian or Pacific Islander.

I could, of course, be completely wrong in my preconceived notions of the ethnic and racial makeup of the Wharton student body, in which case please accept my apologies.

See `with(upenn, table(race.upenn, countryofbirth, useNA = 'always'))` for details.
```
          countryofbirth
race.upenn  1  2 <NA>
      1    29  0    0
      2    13  1    0
      3    17 13    0
      4     5  1    0
      5     7  0    0
      <NA>  0  0    0
```

**Comment by raklein** [comment]
**EDIT: I took a closer look at upenn and also audited the rest of the In House race codings. Below in bold are my notes on just a few changes that are still required. Also, I see the confusion with upenn, as the correct coding is hidden in their notes as opposed to the recoded data or the qualtrics file.**

All In House sites may have custom coding for "race" (and potentially other variables -- so they need to be checked before merging into the primary "race" column). Teams were asked to recode to match the template but this did not _always_ happen. All In House materials can be found in this component: https://osf.io/drfg2/

**upenn: In the case of Wharton, they note in their Readme (https://osf.io/8atc9/) the following coding:
For race:
1-White/Caucasian
2-Black/African American
3-Asian/Pacific Islander
4-Hispanic/Latin American
5-Other**

**For political orientation:
1-Left
2-Right
3-Other**

**ufl (needs recoded -- note this was run by a colleague at UF)
			{text:'American Indian/Alaska Native',value:1},
			{text:'East Asian',value:2},
			{text:'South Asian',value:3},
			{text:'Native Hawaiian or other Pacific Islander',value:4},
			{text:'Black or African American',value:5},
			{text:'White',value:6},
			{text:'Other or Unknown',value:7}**

**wpi (already recoded in raw data -- I previously renamed their race variable to "race.wpi" but we can change it back to "race" to match data template. Also, I see a note about their countryofbirth variables -- note 'countryofbirth' is already recoded so no action needed there.)**

All the rest are good already:
ithaca (raw data already recoded before sending to us)  
wesleyan_inhouse (raw data already recoded before sending to us)  
uwmadison_inhouse (already recoded in Joe's branch)  
sou_inhouse (already recoded in raw data)  
plu (already recoded in raw data, oddities treated in Joe's branch)  
azusa (already recoded in raw data, oddities treated in Joe's branch)  
illinois (already treated in Joe's branch)  
kansas_inhouse (already treated in Joe's branch)  
pace_inhouse (already treated in Joe's branch)

**Comment by raklein** [comment]
@Joe-Hilgard It looks like upenn was addressed in the recent pull, but UFL may still need recoded to match the template? And now that we've verified the coding for wpi, I think we can leave their race variable labelled 'race' (instead of the custom 'race.wpi').

**Comment by Joe-Hilgard** [comment]
Thanks for confirming that the race/ethnicity/countryofbirth variables from WPI were already recoded. Those variables will now just be left as `race`, `ethnicity`, and `birthcountry`.

I've now made my recoding of the UFL race demographics. 
UFL 1 is mapped to template 3 ("American Indian / Alaska Native").
UFL 2 and 3 are mapped to template 4 ("East Asian" & "South Asian" are now "Asian")
UFL 4 is now mapped to template 5 ("Native Hawaiian or other Pacific Islander")
UFL 5 is now mapped to template 2 ("Black or African American")
UFL 6 is now mapped to template 1 ("White")
UFL 7 is now mapped to template 6 ("Something else")

UFL does not appear to have data on hispanic/latino ethnicity.

This is now resolved in commit https://github.com/Joe-Hilgard/ml4/commit/4053727e476c3faa598cdf8f956d42c9b82cc55c
